PSMDB-1033 Documented AWS IAM authentication support (#670)
percona/psmdb-docs: docs/aws-iam.md

# AWS IAM authentication

!!! admonition "Version added: [4.4.19-19](release_notes/4.4.19-19.md)"

!!! important

    This is a [technical preview feature](glossary.md#technical-preview-feature)

IAM (Identity Access Management) is the AWS service that allows you to securely control access to AWS resources. Percona Server for MongoDB supports authentication with AWS IAM enabling you to use the same AWS credentials both for it and other components of your infrastructure. This saves your DBAs from managing different sets of secrets and frees their time on other activities. 

> You can configure AWS IAM for a password-less authentication. Instead of username and password, the user or the application presents the AWS security credentials for authentication, but the secret key is not sent to Percona Server for MongoDB thus significantly increasing the security in your infrastructure. For this scenario, configure the AWS instance that you use to support temporary credentials.

Reviewer: ```suggestion
You can configure AWS IAM for a password-less authentication. Instead of username and password, the user or the application presents the AWS security credentials for authentication, but the secret key is not sent to Percona Server for MongoDB thus significantly increasing the security in your infrastructure.
```
I don't get this sentence, either remove it or elaborate for which exactly scenario?

Author: Removed. I meant to use temp creds for passwordless scenario

> You can configure AWS IAM for a password-less authentication. Instead of username and password, the user or the application presents the secret key and the temporary credentials for authentication. The secret key is not sent to Percona Server for MongoDB thus significantly increasing the security in your infrastructure. For this scenario, configure the AWS instance that you use to support temporary credentials.

Reviewer: user or the application presents AWS security credentials for authentication, but the secret key is not sent to Percona Server for MongoDB thus significantly increasing the security in your infrastructure. 

Please remove this sentence:
For this scenario, configure the AWS instance that you use to support temporary credentials.

> You can configure AWS IAM for a password-less authentication. Instead of username and password, the user or the application presents the secret key and the temporary credentials for authentication. The secret key is not sent to Percona Server for MongoDB thus significantly increasing the security in your infrastructure. For this scenario, configure the AWS instance that you use to support temporary credentials.

Reviewer: ```suggestion
You can configure AWS IAM for a password-less authentication. Instead of username and password, the user or the application presents AWS security credentials for authentication, but the secret key is not sent to Percona Server for MongoDB thus significantly increasing the security in your infrastructure.
```

Author: Done

Percona Server for MongoDB supports two authentication types: 

## User authentication

This authentication type is typically used by human operators. Every user account in AWS has the ARN (Amazon Resource Name), which uniquely identifies this account and the user associated with it. During authentication, the ARN is used to verify the user's identity. 

## Role authentication

> This type is typically used for applications / `mongo` clients. The application must be running on AWS services like EC2 instance or ECS (Elastic Container Service) which uses the IAM role assigned to it. This IAM role ARN is used to authenticate the user in Percona Server for MongoDB.  

Reviewer 2: `mongo` shell is deprecated. If this is about shell then use `mongosh` instead.

> This type is typically used for applications / `mongo` clients. The application must be running on AWS services like EC2 instance or ECS (Elastic Container Service) which uses the IAM role assigned to it. This IAM role ARN is used to authenticate the user in Percona Server for MongoDB.  

Reviewer: ```suggestion
This type is typically used for applications. The application must be running on AWS resource like EC2 instance or ECS (Elastic Container Service) which uses the IAM role assigned to it. This IAM role ARN is used to authenticate the user in Percona Server for MongoDB.  
```

Author: Fixed

> This type is typically used for applications / `mongo` clients. For instance, if your application is running on AWS resources like EC2 instance or ECS (Elastic Container Service) which uses the IAM role assigned to it. Another scenario is to allow users to assume the IAM role and in such a way, grant a user the permissions outlined in the IAM role. The ARN of the IAM role is used to authenticate the application in Percona Server for MongoDB.  

Reviewer: ```suggestion
This type is typically used for applications. For instance, if your application is running on AWS resources like EC2 instance or ECS (Elastic Container Service) which uses the IAM role assigned to it. Another scenario is to allow users to assume the IAM role and in such a way, grant a user the permissions outlined in the IAM role. The ARN of the IAM role is used to authenticate the application in Percona Server for MongoDB.  
```

Really appreciate including scenario with assume role ♡
